modules/drivers/druid/src/metabase/driver/druid/client.clj

```diff
@@ -107,23 +107,18 @@
   [canceled-chan details query]
   {:pre [(map? details) (map? query)]}
   (let [query-id  (get-in query [:context :queryId])
-        query-fut (future
-                    (try
-                      (do-query details query)
-                      (catch Throwable e
-                        e)))
-        cancel! (delay
-                  (cancel-query-with-id! details query-id))]
+        ;; Run the query in a future so that we can interrupt it from the canceled-chan callback.
+        query-fut (future (do-query details query))
+        undo! #(undo-query-with-id! details query-id)]
     (a/go
       (when (a/<! canceled-chan)
         (future-cancel query-fut)
-        @cancel!))
-    (try
-      ;; Run the query in a future so that this thread will be interrupted, not the thread running the query (which is
-      ;; not interrupt aware)
-      (u/prog1 @query-fut
-        (when (instance? Throwable <>)
-          (throw <>)))
-      (catch InterruptedException e
-        @cancel!
-        (throw e)))))
+        (undo!)))
+    (try @query-fut
+         (catch java.util.concurrent.ExecutionException e
```

Comment:
However, after the future-internal exceptions are rethrown, they are wrapped in `ExecutionException`. To preserve the existing behavior, that wrapper is removed here.